components/script/dom/bindings/str.rs

```diff
@@ -303,14 +303,20 @@ impl DOMString {
         parse_week_string(&*self.0).is_ok()
     }
 
-    /// A valid number is the same as what rust considers to be valid,
-    /// except for +1., NaN, and Infinity.
     /// https://html.spec.whatwg.org/multipage/#valid-floating-point-number
-    pub fn is_valid_number_string(&self) -> bool {
-        let input = &self.0;
-        input.parse::<f64>().ok().map_or(false, |val| {
-            !(val.is_infinite() || val.is_nan() || input.ends_with(".") || input.starts_with("+"))
-        })
+    pub fn is_valid_floating_point_number_string(&self) -> bool {
+        // for the case that `parse_floating_point_number` cannot handle
+        if *&self.0.contains(" ") {
```

Comment:
`valid floating number` cannot include whitespace.
skipping the front whitespace is for parsing, which is for making the input to `best presentation`.

the two purpose are different. One can have whitespace and the one cannot. That's why the code here do this.

Reply:
This defies intuition, since this would mean an invalid floating point number string can successfully be parsed. I'm going to have to clarify the spec language with WHATWG.

Regardless of what the result may be, you still need to address the first question of my last comment.

Reply:
Okay, so WHATWG replied that it is indeed OK to successfully parse an invalid string. Ignore that last comment and address the first question in the first comment.

Reply:
@KiChjang  fixed the first comment

Reply:
@KiChjang  fixed